tests/ClientTest.php

```diff
@@ -332,24 +332,27 @@ public function testAddConnectionWithConcreteWriteStream()
 
     /**
      * Tests that adding a connection configured to use the SSL transport and
-     * force use of IPv4 throws an exception.
+     * force use of IPv4 does not throw an exception.
      *
      * @see https://github.com/reactphp/socket-client/issues/4
      */
-    public function testAddConnectionWithSslTransportAndForceIpv4ThrowsException()
+/*
+    public function testAddConnectionWithSslTransportAndForceIpv4()
     {
         $connection = $this->getMockConnectionForAddConnection();
         Phake::when($connection)->getOption('transport')->thenReturn('ssl');
         Phake::when($connection)->getOption('force-ipv4')->thenReturn(true);
+        $writeStream = $this->getMockWriteStream();
+        Phake::when($this->client)->getWriteStream($connection)->thenReturn($writeStream);
+
+        $this->client->setLogger($this->getMockLogger());
         $this->client->setResolver($this->getMockResolver());
+        $this->client->addConnection($connection);
```

Comment:
I think it's sufficient for this test to simply call `addConnection()` as above. So long as an exception is no longer thrown, that's all the test should need to verify. I'm skeptical that it would be worth the effort to test for lower-level conditions beyond that.